api/client.go

```diff
@@ -146,6 +146,10 @@ func (c Client) REST(method string, p string, body io.Reader, data interface{})
 		return handleHTTPError(resp)
 	}
 
+	if resp.StatusCode == http.StatusNoContent && resp.ContentLength == 0 {
```

Comment:
I think that checking `http.StatusNoContent` is enough for this condition; we know that the 204 response doesn't contain a body and so there's no need to also check `resp.ContentLength == 0`. Would you agree?

Reply:
No disagreement. I was trying to protect against outliers based on previous experience, as the current implementation prevents the clients from the ability to detect when a service is effectively returning conflicting results. Will make it less restrictive.